File: backend/src/models/aiModels.ts
export type ContentType = 'URL' | 'MESSAGE' | 'QR_TEXT' | 'OCR_TEXT';
export type RiskLevel = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL' | 'UNKNOWN';

export interface AiIndicator {
    title: string;
    description: string;
}

export interface AiAnalysisRequest {
    contentType: ContentType;
    content: string;
    localRiskScore: number;
    localRiskLevel: string;
    localIndicators: string[];
}

export interface AiAnalysisResponse {
    riskLevel: RiskLevel;
    riskScore: number;
    summary: string;
    indicators: AiIndicator[];
    recommendation: string;
    confidence: number;
    aiSource: 'gemini' | 'fallback';
}

File: app/src/main/java/com/marwadiuniversity/trustlens/domain/model/AiModels.kt
package com.marwadiuniversity.trustlens.domain.model

enum class AiRiskLevel {
    LOW,
    MEDIUM,
    HIGH,
    CRITICAL,
    UNKNOWN
}

data class AiIndicator(
    val title: String,
    val description: String
)

data class AiAnalysisRequest(
    val contentType: String,
    val content: String,
    val localRiskScore: Int,
    val localRiskLevel: String,
    val localIndicators: List<String>
)

data class AiAnalysisResult(
    val riskLevel: AiRiskLevel,
    val riskScore: Int,
    val summary: String,
    val indicators: List<AiIndicator>,
    val recommendation: String,
    val confidence: Float,
    val aiSource: String = "fallback"
)
